### ISSUE TITLE
specimin crashes in java parser for cf-3020

### ISSUE THREAD

**Issue by tahiat** [kelloggm/specimin#222]
Issue link [cf-3020](https://github.com/typetools/checker-framework/issues/3020)
[actual code](https://github.com/google/guava/blob/33bdb36c7943f93d2e158fc4e9c115cafbd4ab02/guava/src/com/google/common/util/concurrent/FuturesGetChecked.java#L191)

The linked [source code](https://github.com/google/guava/blob/28b36c80447729d1edeffedd70dc4a7ac91e76ce/guava/src/com/google/common/util/concurrent/FuturesGetChecked.java#L189) in the issue does not reproduce the bug.  [This](https://github.com/google/guava/blob/33bdb36c7943f93d2e158fc4e9c115cafbd4ab02/guava/src/com/google/common/util/concurrent/FuturesGetChecked.java#L191)  should generate the bug (with small modification). However, specimin crashes in javaparser when minimizing. 
command:  `./gradlew run --args='--outputDirectory "/Users/tahiatul/Documents/test/NULLNESS/output" --root "/Users/tahiatul/Documents/test/NULLNESS/guava/guava/src/" --targetFile "com/google/common/util/concurrent/FuturesGetChecked.java" --targetMethod "com.google.common.util.concurrent.FuturesGetChecked.GetCheckedTypeValidatorHolder#getBestValidator()"'`

stack traces:

```
> Configure project :
No explicit dependency on the Checker Framework found, using default version 3.42.0

> Task :run
Exception in thread "main" com.github.javaparser.ParseProblemException: (line 4,col 73) Parse error. Found "?", expected one of  ")" "@" "abstract" "boolean" "byte" "char" "default" "double" "enum" "exports" "final" "float" "int" "long" "module" "native" "open" "opens" "private" "protected" "provides" "public" "record" "requires" "short" "static" "strictfp" "synchronized" "to" "transient" "transitive" "uses" "volatile" "with" "yield" <IDENTIFIER>
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13708)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13554)
  com.github.javaparser.GeneratedJavaParser.Parameters(GeneratedJavaParser.java:2236)
  com.github.javaparser.GeneratedJavaParser.MethodDeclaration(GeneratedJavaParser.java:2026)
  com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBodyDeclaration(GeneratedJavaParser.java:1669)
  com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBody(GeneratedJavaParser.java:1183)
  com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceDeclaration(GeneratedJavaParser.java:500)
  com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:152)
  com.github.javaparser.JavaParser.parse(JavaParser.java:123)
  com.github.javaparser.JavaParser.parse(JavaParser.java:232)
  com.github.javaparser.StaticJavaParser.parse(StaticJavaParser.java:167)
  org.checkerframework.specimin.SpeciminRunner.parseJavaFile(SpeciminRunner.java:473)
  org.checkerframework.specimin.SpeciminRunner.performMinimization(SpeciminRunner.java:293)
  org.checkerframework.specimin.SpeciminRunner.main(SpeciminRunner.java:77)

        at com.github.javaparser.StaticJavaParser.handleResult(StaticJavaParser.java:270)
        at com.github.javaparser.StaticJavaParser.parse(StaticJavaParser.java:167)
        at org.checkerframework.specimin.SpeciminRunner.parseJavaFile(SpeciminRunner.java:473)
        at org.checkerframework.specimin.SpeciminRunner.performMinimization(SpeciminRunner.java:293)
        at org.checkerframework.specimin.SpeciminRunner.main(SpeciminRunner.java:77)

> Task :run FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':run'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk-11.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
```

**Comment by LoiNguyenCS** [comment]
Hello @tahiat,

So is [this](https://github.com/google/guava/tree/33bdb36c7943f93d2e158fc4e9c115cafbd4ab02) the input source code that we need to look into?

**Comment by tahiat** [comment]
@LoiNguyenCS  this [one](https://github.com/google/guava/tree/nullness-checker-framework)

**Comment by LoiNguyenCS** [comment]
sorry, this is confusing. So which branch are we interested in?

**Comment by LoiNguyenCS** [comment]
It should be resolved by #223

**Comment by LoiNguyenCS** [comment]
Resolved by #223
